### ISSUE TITLE
Weird image freezing when launching and moving the window

### ISSUE THREAD

**Issue by Lau-San** [ppy/osu#29487]
### Type

Other

### Bug description

On Linux Hyprland(wayland) I used to be able to run Osu! Lazer without issue, out of the box. Now, after updating both Osu! and Hyprland (don't remember the previous working versions), here's what happens:

- I launch Osu!, it starts, but it gets stuck on like the first frame (a black screen and fps counter) but only the image. I can hear everything. I can even click on buttons and "do stuff" but I can't see anything.
- I switch to windowed, same problem.
- I resize the window, suddenly, it works for some reason and I can see the image updating as it should. I can actually see what I'm doing, etc.
- I move the window, it freezes again.
- I switch to fullscreen, it freezes.
- I switch to borderless, it freezes.

So basically, Every time I launch the game, move the window, maximize, or switch to fullscreen, the image freezes but the rest of the game is still running. Sound is still going, buttons are working, I'm just stuck looking at a still frame, trying to guess where my cursor is. UNLESS the last thing I did with the window was resize it for some reason.

### Screenshots or videos

Forgot to record the audio but know audio works fine at all times. I can hear the music, the hover and click sound effects, everything.

https://github.com/user-attachments/assets/d7f955ef-627e-41e2-aa8e-46e04e7601cb

### Version

2024.817.0

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/16646755/compressed-logs.zip)

**Comment by bdach** [comment]
This should 100% not be reported here, and 99% reported to hyprland instead (1% chance it's an sdl issue). Likely to be caused by them removing wlroots as a dependency.
